servers/physics_2d/space_2d_sw.cpp

```diff
@@ -81,6 +81,9 @@ int Physics2DDirectSpaceStateSW::intersect_point(const Vector2& p_point,ShapeRes
 		if (!shape->contains_point(local_point))
 			continue;
 
+		if (cc>=p_result_max)
+			continue;
```

Comment:
Might have been better with `break` or even in the for loop condition. Should I make a PR for this? :smiley: